enterprise/cypher/cypher/src/main/scala/org/neo4j/cypher/internal/compatibility/v3_3/runtime/PipelineInformation.scala

```diff
@@ -20,15 +20,113 @@
 
 package org.neo4j.cypher.internal.compatibility.v3_3.runtime
 
+import org.neo4j.cypher.internal.compiler.v3_3.planner.logical.plans.LogicalPlan
 import org.neo4j.cypher.internal.frontend.v3_3.InternalException
 import org.neo4j.cypher.internal.frontend.v3_3.symbols.CypherType
 
+import scala.collection.mutable
 
 object PipelineInformation {
   def empty = new PipelineInformation(Map.empty, 0, 0)
 
   def apply(slots: Map[String, Slot], numberOfLongs: Int, numberOfReferences: Int) =
     new PipelineInformation(slots, numberOfLongs, numberOfReferences)
+
+  def toString(startFrom: LogicalPlan, m: Map[LogicalPlan, PipelineInformation]): String = {
```

Comment:
You are right. It helped me debug a problem, and I thought it was helpful enough to stick around, but I'm not sure how to expose it to the public yet.